docs/firestore/pagination.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Pagination
+description: Pagination example using Cloud Firestore.
+next: /functions/usage
+previous: /firestore/usage-with-flatlists
+---
+
+Pagination using cloud firestore can be done in various ways but here's the very basic way to do it.
+We are going to make the pagination using querying features provided by firestore.
+[`orderBy`, `limit`, `startAfter`]
+
+# Setup state
+
+First, setup a component which will display the list of data.
+The component will have 2 separate states; `last` and `userData`:
+
+```jsx
+import React, {useState} from 'react';
+import type {Node} from 'react';
+import {Text, View, Button, Alert} from 'react-native';
+
+import firestore from '@react-native-firebase/firestore';
+
+const userCollection = firestore().collection('Users');
+
+const App: () => Node = () => {
+  const [lastSnapshot, setLastSnapshot] = useState(null);
+  const [userData, setUserData] = useState([]);
+};
+```
+
+# `LoadData` function
+
+Next, we'll make a function called `LoadData` that fetches data from `Users` collection.
+This function will be hooked with the `Button`.
+
+If lastSanpshot is not assigned (meaning initial load), the function will fetch from the start.
+After successful fetch from the collection, store the last snapshot data by `setLastSnapshot`.
```

Comment:
```suggestion
If lastDocument is undefined the function will fetch from the star to perform the initial loadt, otherwise it use the lastDocument from the snapshot
After successful fetch from the collection, store the last document we accessed with `setLastDocument`.
```